src/transaction.js

```diff
@@ -154,6 +146,11 @@ class Transaction {
   }
 
   cleanup() {
+    // Don't release the connection if there's a parent transaction or
+    // if we've already cleaned up
+    if (this.parent || this.connection.uuid === undefined) return;
+
+    this._clearCls();
```

Comment:
I see you removed two calls to `this._clearCls();` above and added one here, why?

Reply:
I think it makes more sense to put it here, because it ensures that we do the cleanup even if neither `rollback` nor `commit` are called. This is needed for the MySQL cases.

1. User calls `sequelize.transaction`
2. MySQL throws for a deadlock. It marks `transaction.finished = 'rollback'`
3. In sequelize.js, since the transaction is already finished, we call `transaction.cleanup` without calling `transaction.rollback`

If we only called `_clearCls()` in `rollback` or `commit`, it wouldn't get called in this case, and the CLS (Continuation Local Storage) wouldn't get cleared.

It also makes the `cleanup` function more closely mirror the `prepareEnvironment` function. I added a comment mentioning that too!